src/dof_management/freedom_signature.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <type_traits>
+
+#include <Kokkos_Core.hpp>
+
+namespace openturbine {
+
+enum class FreedomSignature : std::uint8_t {
+    AllComponents = 0b00111111,
+    JustPosition = 0b00111000,
+    JustRotation = 0b00000111,
+    NoComponents = 0b00000000
+};
+
+KOKKOS_INLINE_FUNCTION
+FreedomSignature operator|(FreedomSignature x, FreedomSignature y) {
+    using T = std::underlying_type_t<FreedomSignature>;
+    return static_cast<FreedomSignature>(static_cast<T>(x) | static_cast<T>(y));
+}
+
+KOKKOS_INLINE_FUNCTION
+size_t count_active_dofs(FreedomSignature x) {
+    using T = std::underlying_type_t<FreedomSignature>;
+    auto count = 0UL;
+    constexpr auto zero = T{0};
+    constexpr auto one = T{1};
+    for (auto value = static_cast<T>(x); value > zero; value = value >> 1) {
```

Comment:
It's a shame that https://en.cppreference.com/w/cpp/numeric/popcount is in C++20, it would be great here.

Reply:
Agreed. I initially prototyped this this bitset's count function, which does the same thing. Sadly, we can't expect to be able to use the standard library on device, so we have to roll our own.